app/services/manifest_token_json_converter.rb

```diff
@@ -0,0 +1,22 @@
+class ManifestTokenJsonConverter
```

Comment:
Better, yet, I would mangle the params in the controller. That's really what a controller is for: translating user input into something the model can act upon.